jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -2482,6 +2507,164 @@ ecma_string_trim (const ecma_string_t *string_p) /**< pointer to an ecma string
   return ret_string_p;
 } /* ecma_string_trim */
 
+/**
+ * Create a string builder
+ *
+ * @return new string builder
+ */
+ecma_stringbuilder_t
+ecma_stringbuilder_create (void)
+{
+  const lit_utf8_size_t initial_size = sizeof (ecma_string_t);
+  ecma_stringbuilder_header_t *buffer_p = (ecma_stringbuilder_header_t *) jmem_heap_alloc_block (initial_size);
+  buffer_p->current_size = initial_size;
+
+  ecma_stringbuilder_t ret;
+  ret.header_p = buffer_p;
+
+  return ret;
+} /* ecma_stringbuilder_create */
```

Comment:
I'd suggest to have a function to create a string builder with an ecma-string argument. Therefore one less reallocation is needed. It could be useful in array/typedarray builtin routines.